python_modules/dagster/dagster/core/instance/__init__.py

```diff
@@ -1316,7 +1336,13 @@ def report_engine_event(
 
         check.class_param(cls, "cls")
         check.str_param(message, "message")
-        check.inst_param(pipeline_run, "pipeline_run", PipelineRun)
+        check.opt_inst_param(pipeline_run, "pipeline_run", PipelineRun)
+        check.opt_str_param(run_id, "run_id")
+        check.opt_str_param(pipeline_name, "pipeline_name")
+
+        run_id = run_id if run_id else pipeline_run.run_id
```

Comment:
are we sure that we will never get here with run_id=None and pipeline_run=None? (similar comment for the line below)